src/tasks/actions/thunks.ts

```diff
@@ -244,15 +247,55 @@ export const deleteTask = (taskID: string) => async (
   }
 }
 
-export const cloneTask = (task: Task) => async (dispatch: Dispatch<Action>) => {
+export const cloneTask = (task: Task) => async (
+  dispatch: Dispatch<Action>,
+  getState: GetState
+) => {
   try {
+    const state = getState()
     const resp = await api.getTask({taskID: task.id})
 
     if (resp.status !== 200) {
       throw new Error(resp.data.message)
     }
 
-    const newTask = await api.postTask({data: resp.data})
+    const taskName = resp.data.name
+    const tasks = getAll<Task>(state, ResourceType.Tasks)
+    const allTaskNames = tasks.map(d => d.name)
+    const clonedName = incrementCloneName(allTaskNames, taskName)
+    const {flux} = resp.data
+
+    const ast = parse(flux)
+
+    ast.body = ast.body.map(statement => {
+      if (
+        statement.type === 'OptionStatement' &&
```

Comment:
i think you also have to look for the option to be named `task` somewhere in here, or else it'll overwrite the variable `name` if:
```
option v = {
    name: 'yolo'
}

option task = {
    name: 'task to clone'
}
````

Reply:
I think that's illegal in some states but better put up some guards